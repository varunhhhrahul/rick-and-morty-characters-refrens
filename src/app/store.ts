import { configureStore, Action, getDefaultMiddleware } from "@reduxjs/toolkit";
import { ThunkAction } from "redux-thunk";

import rootReducer, { RootState } from "./rootReducer";

const store = configureStore({
  reducer: rootReducer,
  middleware: () => [...getDefaultMiddleware()],
  // middleware: [...getDefaultMiddleware({ immutableCheck: false })],
});

export type AppDispatch = typeof store.dispatch;

export type AppThunk = ThunkAction<void, RootState, null, Action<string>>;

export default store;
