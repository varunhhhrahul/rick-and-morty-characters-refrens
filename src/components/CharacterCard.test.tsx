import React from "react";
import {
  waitFor,
  screen,
  waitForElementToBeRemoved,
  act,
} from "@testing-library/react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import store from "../app/store";
import CharactedCard from "./CharacterCard";

const character = {
  created: new Date("2017-11-04T18:48:46.250Z"),
  episode: [
    "https://rickandmortyapi.com/api/episode/1",
    "https://rickandmortyapi.com/api/episode/2",
  ],
  gender: "Male",
  id: 1,
  image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
  location: {
    name: "Citadel of Ricks",
    url: "https://rickandmortyapi.com/api/location/3",
  },
  name: "Rick Sanchez",
  origin: {
    name: "Earth (C-137)",
    url: "https://rickandmortyapi.com/api/location/1",
  },
  species: "Human",
  status: "Alive",
  type: "",
  url: "https://rickandmortyapi.com/api/character/1",
};
let component: renderer.ReactTestRenderer | null = null;
describe("CharacterCard", () => {
  beforeEach(async () => {
    await act(async () => {
      component = renderer.create(
        <Provider store={store}>
          <CharactedCard character={character} />
        </Provider>
      );
    });
  });

  afterEach(() => {
    component = null;
  });

  test("should render correctly", async () => {
    const tree = component!.toJSON();

    expect(tree).toMatchSnapshot();
  });
  test("should render correctly with image", async () => {
    await waitFor(() =>
      expect(component!.root.findByType("img").props.src).toBe(character.image)
    );
  });
  test("should render title correctly", async () => {
    await waitFor(() =>
      expect(
        component!.root.findAllByType("h6")[0].children.includes(character.name)
      ).toBe(true)
    );
  });
});
