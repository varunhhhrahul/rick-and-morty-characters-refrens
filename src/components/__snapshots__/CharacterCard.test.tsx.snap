// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard should render correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5abhjx-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiBox-root css-0"
  >
    <div
      className="MuiBox-root css-r3okx0"
    >
      <img
        alt="Rick Sanchez"
        className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-6mo23z-MuiCardMedia-root"
        height="200"
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
    </div>
    <div
      className="MuiCardContent-root css-b26uya-MuiCardContent-root"
    >
      <div
        className="MuiBox-root css-vxgrp0"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-4evpfe-MuiTypography-root"
        >
          Rick Sanchez
        </h6>
        <div
          className="MuiBox-root css-0"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 css-15ubpqm-MuiTypography-root"
          >
            <div
              style={
                Object {
                  "backgroundColor": "green",
                  "borderRadius": 50,
                  "height": 10,
                  "marginRight": 5,
                  "width": 10,
                }
              }
            />
             
            Alive
          </h6>
        </div>
      </div>
      <div
        className="MuiBox-root css-vxgrp0"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
        >
          Human
           - 
          Male
        </h6>
      </div>
      <div
        className="MuiBox-root css-0"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Origin
        </h6>
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1se7cm8-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
      </div>
      <div
        className="MuiBox-root css-0"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Current Location
        </h6>
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1se7cm8-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
      </div>
      <div
        className="MuiBox-root css-0"
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Episodes
          </h6>
        </div>
        <br />
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1se7cm8-MuiSkeleton-root"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
